File: docs/src/neetcode150/go/top_k_frequent_element.go
package main

func topKFrequent(nums []int, k int) []int {
	freq := make(map[int]int)
	for _, num := range nums {
		freq[num]++
	}
	buckets := make(map[int][]int)
	for num, count := range freq {
		buckets[count] = append(buckets[count], num)
	}

	var result []int
	for i := len(nums); i >= 0; i-- {
		vals, ok := buckets[i]
		if ok {
			result = append(result, vals...)
		}
	}
	return result[:k]
}

File: docs/src/neetcode150/py/top_k_frequent_element.py
from typing import List
from collections import Counter, defaultdict
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freq = Counter(nums)
        buckets = defaultdict(list)
        for num, count in freq.items():
            buckets[count].append(num)

        result = []
        for count in range(len(nums), 0, -1):
            if count in buckets:
                result.extend(buckets[count])

        return result[:k]